packages/toolpad-app/src/toolpad/AppEditor/PagesExplorer/index.tsx

```diff
@@ -250,8 +329,65 @@ export default function PagesExplorer({ className }: PagesExplorerProps) {
           aria-level={1}
           labelText="Pages"
           createLabelText="Create page"
-          onCreate={handleCreatePageDialogOpen}
+          onCreate={handleCreateNewPage}
         >
+          {isCreateNewPageOpen ? (
```

Comment:
Between this and https://github.com/mui/mui-toolpad/pull/2690 is there any opportunity to share logic? I feel like in several places we need an elaborate tree component with create/edit support

Reply:
Agreed on the need to have a common tree component for our "explorer" use cases with the editor. It'll take some planning and dedicated effort to bridge the gap, since currently even the components for the Hierarchy explorer and the (proposed) Queries explorer are separate. Does it make sense to create an issue to track merging all tree components into one?

Reply:
I've created a `CreateTreeItem` component for now - should be able to use it in the global functions explorer too.

Reply:
> Does it make sense to create an issue to track merging all tree components into one?

If we want to create a whole tree view component with our own setting and features to reuse everywhere we can create a separate issue for it and do it, feels a bit early for this PR but should be a good idea long term.

Reply:
Adding it as a discussion item for next to next grooming